python/ray/train/mosaic/mosaic_trainer.py

```diff
@@ -0,0 +1,168 @@
+import inspect
+import os
+from typing import TYPE_CHECKING, Callable, Dict, Optional
+
+from composer.trainer import Trainer
+
+from ray.air import session
+from ray.train.constants import (
+    EVALUATION_DATASET_KEY,
+    TRAIN_DATASET_KEY,
+)
+from ray.air.checkpoint import Checkpoint
+from ray.air.config import DatasetConfig, RunConfig, ScalingConfig
+from ray.train.torch import TorchConfig, TorchTrainer
+from ray.train.trainer import GenDataset
+from ray.util import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.data.preprocessor import Preprocessor
+
+
+@PublicAPI(stability="alpha")
+class MosaicTrainer(TorchTrainer):
+    """A Trainer for data parallel Mosaic Composers on PyTorch training.
+
+    This Trainer runs the ``composer.trainer.Trainer.fit()`` method on multiple
+    Ray Actors. The training is carried out in a distributed fashion through PyTorch
+    DDP. These actors already have the necessary torch process group already
+    configured for distributed PyTorch training.
+
+    The training function ran on every Actor will first run the
+    specified ``trainer_init_per_worker`` function to obtain an instantiated
+    ``composer.Trainer`` object. The ``trainer_init_per_worker`` function
+    will have access to preprocessed train and evaluation datasets.
+
+    Example:
+        .. code-block:: python
```

Comment:
do we still need this code-block:: python here?

Reply:
I'm not sure. Where can I find more information regarding doc string formatting?

Reply:
Yeah I think we should remove it, but once we add it to the API ref we can confirm if it's being rendered correctly.